diff --git a/tweettoot.py b/tweettoot.py
index e621c8d..ea2ec6d 100644
--- a/tweettoot.py
+++ b/tweettoot.py
@@ -64,7 +64,8 @@ class TweetToot:
         # Only initialize the Mastodon API if we find something
         mastodon_api = Mastodon(
             access_token=self.mastodon_token,
-            api_base_url=self.mastodon_url
+            api_base_url=self.mastodon_url,
+            feature_set="pleroma"
         )
 
         media_ids = []
@@ -181,6 +182,7 @@ class TweetToot:
             media_ids = media_ids,
             visibility = "public",
             sensitive = False,
+            content_type = "text/html",
             spoiler_text = None,
             in_reply_to_id = None
         )
@@ -194,6 +196,6 @@ class TweetToot:
     def escape_usernames(self, text):
         ats = re.findall(self.twitter_username_re, text)
         for at in ats:
-            text = text.replace("@"+at, "@*"+at)
+            text = text.replace("@"+at, "<a href=\"https://twitter.com/"+at+"\">@"+at+"@twitter.com</a>")
 
         return text
